src/Command/GenerateCommand.php

```diff
@@ -0,0 +1,383 @@
+        if (
+            !is_dir(Path::join($dataDir, CloneCommand::DIR_CLONE, BreakingChangesComparer::FROM))
+            || !is_dir(Path::join($dataDir, CloneCommand::DIR_CLONE, BreakingChangesComparer::TO))
+        ) {
+            throw new InvalidOptionException(
+                "'$dataDir' is missing one or both of the cloned directories. Run the clone command."
+            );
+        }
+
+        $fromFile = Path::join($dataDir, CloneCommand::DIR_CLONE, BreakingChangesComparer::FROM, CloneCommand::META_FILE);
+        $toFile = Path::join($dataDir, CloneCommand::DIR_CLONE, BreakingChangesComparer::TO, CloneCommand::META_FILE);
+
+        $this->metaDataFrom = $this->getJsonFromFile($fromFile);
+        $this->metaDataTo = $this->getJsonFromFile($toFile);
+        $this->metaDataFrom['branch'] = $this->guessBranchFromConstraint($this->metaDataFrom['constraint']);
+        $this->metaDataTo['branch'] = $this->guessBranchFromConstraint($this->metaDataTo['constraint']);
+    }
+
+    /**
+     * Identifies supported modules which are relevant for our purposes
+     */
+    private function findSupportedModules(): void
+    {
+        $this->output->writeln('Finding supported modules across both branches.');
+        $supportedModules = MetaData::getAllRepositoryMetaData(true)[MetaData::CATEGORY_SUPPORTED];
+        $cmsMajorTo = $this->getCmsMajor($this->metaDataTo);
+        // Make sure we only have supported modules that are in the "to" version.
+        // Don't also restrict to the "from" version because we need to know about new modules for API links.
+        $supportedModules = MetaData::removeReposNotInCmsMajor($supportedModules, $cmsMajorTo, true);
+        $this->supportedModules = $supportedModules;
+    }
+
+    /**
+     * Determine which major release line of Silverstripe CMS this metadata applies to
+     */
+    private function getCmsMajor(array $metaData): string
+    {
+        $composerJson = $this->getJsonFromFile(Path::join($metaData['path'], 'composer.json'), false);
+        // Note that while the branch will usually directly tell us what major it is since our recipes
+        // tend to be lockstepped, we have an API to explicitly fetch it so we may as well for future proofing.
+        return BranchLogic::getCmsMajor(
+            MetaData::getMetaDataByPackagistName($metaData['recipe']),
+            $metaData['branch'],
+            $composerJson
+        );
+    }
+
+    /**
+     * Parse the PHP code in the supported modules.
+     * If there are errors, they're stored in $this->parseErrors.
+     * The parsed result is represented by the returned project.
+     */
+    private function parseModules(string $dataDir): Project
+    {
+        $this->output->writeln('Parsing modules...');
+
+        $factory = new ParserFactory($this->supportedModules, $dataDir);
+        $project = $factory->buildProject();
+
+        // @TODO can use callback arg to output current step
```

Comment:
Remove all the todo's

Reply:
I've replaced the todo comments with https://github.com/silverstripe/deprecation-checker/issues/5